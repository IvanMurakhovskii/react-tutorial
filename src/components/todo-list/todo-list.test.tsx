/**
 * @jest-environment jsdom
*/

import React from 'react';
import { mount } from "enzyme";
import TodoList, { TodoListProps } from './todo-list';
import TodoListItem from '../todo-list-item';
import OrderEnum from '../../emums/order-enum';

const onDeletedMock = jest.fn();
const onToggleDoneMock = jest.fn();
const onToggleImportantMock = jest.fn();

const todos = [
  {
    label: 'test 1',
    important: false,
    done: false,
    hidden: false,
    id: 1
  },
  {
    label: 'test 2',
    important: false,
    done: true,
    hidden: false,
    id: 2
  },
  {
    label: 'test 3',
    important: true,
    done: false,
    hidden: false,
    id: 3
  }
]

const props: TodoListProps = {
  todos: todos,
  order: OrderEnum.ASC,
  onDeleted: onDeletedMock,
  onToggleDone: onToggleDoneMock,
  onToggleImportant: onToggleImportantMock
}

const item = <TodoList {...props} />
const wrapper = mount(item);

describe("TodoList", () => {
  it("render item with data", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("lenght of child TodoListItem equals 3", () => {
    expect(wrapper.find(TodoListItem).length).toEqual(3);
  });

});