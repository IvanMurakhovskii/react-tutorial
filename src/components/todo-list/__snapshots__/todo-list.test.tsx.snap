// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoList render item with data 1`] = `
<div
  className="list-group todo-list"
>
  <div
    key="1"
  >
    <Styled(div)>
      <TodoListItem
        done={false}
        hidden={false}
        important={false}
        label="test 1"
        onDeleted={[Function]}
        onToggleDone={[Function]}
        onToggleImportant={[Function]}
      />
    </Styled(div)>
  </div>
  <div
    key="2"
  >
    <Styled(div)>
      <TodoListItem
        done={true}
        hidden={false}
        important={false}
        label="test 2"
        onDeleted={[Function]}
        onToggleDone={[Function]}
        onToggleImportant={[Function]}
      />
    </Styled(div)>
  </div>
  <div
    key="3"
  >
    <Styled(div)>
      <TodoListItem
        done={false}
        hidden={false}
        important={true}
        label="test 3"
        onDeleted={[Function]}
        onToggleDone={[Function]}
        onToggleImportant={[Function]}
      />
    </Styled(div)>
  </div>
</div>
`;
